### PR TITLE
Use @faire/mjml-react react bindings

### PR DESCRIPTION
**Pull request by IanEdington**
## Describe your changes
Use @faire/mjml-react for mjml\<\>react bindings rather than the unmaintained mjml-react.

## Issue link
#203

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.

I've tested it locally by running `yarn dev` and browsing the email templates. This is a very nice tool 🙏 

Most of the tests are passing. 1 test is breaking locally for me. The same test is failing on the main branch locally. Hopefully it passes CI.

* livereload › returns pollers immediatly when vectorClock=0

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -49,7 +50,6 @@
     "@types/jsdom": "^20.0.0",
     "@types/lodash": "^4.14.184",
     "@types/mjml": "^4.7.0",
-    "@types/mjml-react": "^2.0.4",
```
Comment (IanEdington) [inline packages/cli/package.json:52]: types are included in `@faire/mjml-react`

File: yarn.lock [hunk 2]
```diff
@@ -3115,14 +3116,6 @@ babel-preset-jest@^29.0.2:
     babel-plugin-jest-hoist "^29.0.2"
     babel-preset-current-node-syntax "^1.0.0"
 
-babel-runtime@~6.25.0:
```
Comment (IanEdington) [inline yarn.lock:3118]: We removed a number of dependencies that I don't believe mjml-react needs in production. I'd welcome a PR to @Faire/mjml-react if this isn't the case

**Review by alexfarrill — APPROVED** [review]
Looks great!  Thanks for the quick PR.  I ran the livereload test and it passed for me. and we've seen that test be flaky before so that probably explains it.

image

**Review by psugihara — APPROVED** [review]
hell yeah. thanks so much!

cc #203

**Comment by alexfarrill** [comment]
Hey @IanEdington, I had to revert it actually.  The e2e test for a standalone installation (a React project that installs mailing but isn't using a framework like Next, etc.) is not working.  I get this error when running `npx mailing`:

```
Running mailing
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'prop-types'
Require stack:
- /Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/@faire/mjml-react/dist/src/mjml.js
- /Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/@faire/mjml-react/dist/src/index.js
- /Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/mailing/dist/mailing.cjs.dev.js
- /Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/mailing/dist/mailing.cjs.js
- /Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/mailing/bin.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/@faire/mjml-react/dist/src/mjml.js:12:18)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/@faire/mjml-react/dist/src/mjml.js',
    '/Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/@faire/mjml-react/dist/src/index.js',
    '/Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/mailing/dist/mailing.cjs.dev.js',
    '/Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/mailing/dist/mailing.cjs.js',
    '/Users/alexfarrill/projects/mailing_e2e_tests/runs/20220927142254/standalone/node_modules/mailing/bin.js'
  ]
}
e2e_test.rb:190:in `cleanup_io_and_subprocess': No subprocess found to cleanup (RuntimeError)
	from e2e_test.rb:100:in `ensure in block in run'
	from e2e_test.rb:100:in `block in run'
	from e2e_test.rb:84:in `each'
	from e2e_test.rb:84:in `run'
	from e2e_test.rb:209:in `<main>'
/Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:1214:in `__connect_nonblock': Failed to open TCP connection to localhost:3883 (Connection refused - connect(2) for 127.0.0.1:3883) (Errno::ECONNREFUSED)
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:1214:in `connect_nonblock'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:56:in `connect_internal'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:137:in `connect'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:642:in `block in tcp'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:227:in `each'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:227:in `foreach'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:632:in `tcp'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:998:in `connect'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:976:in `do_start'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:965:in `start'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:627:in `start'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:503:in `get_response'
	from /Users/alexfarrill/projects/mailing/scripts/e2e_test/app_configs/app.rb:114:in `wait_for_previews_json!'
	from /Users/alexfarrill/projects/mailing/scripts/e2e_test/app_configs/app.rb:39:in `run_mailing!'
	from e2e_test.rb:94:in `block in run'
	from e2e_test.rb:84:in `each'
	from e2e_test.rb:84:in `run'
	from e2e_test.rb:209:in `<main>'
/Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:1214:in `__connect_nonblock': Connection refused - connect(2) for 127.0.0.1:3883 (Errno::ECONNREFUSED)
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:1214:in `connect_nonblock'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:56:in `connect_internal'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:137:in `connect'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:642:in `block in tcp'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:227:in `each'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:227:in `foreach'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/socket.rb:632:in `tcp'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:998:in `connect'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:976:in `do_start'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:965:in `start'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:627:in `start'
	from /Users/alexfarrill/.asdf/installs/ruby/3.1.2/lib/ruby/3.1.0/net/http.rb:503:in `get_response'
	from /Users/alexfarrill/projects/mailing/scripts/e2e_test/app_configs/app.rb:114:in `wait_for_previews_json!'
	from /Users/alexfarrill/projects/mailing/scripts/e2e_test/app_configs/app.rb:39:in `run_mailing!'
	from e2e_test.rb:94:in `block in run'
	from e2e_test.rb:84:in `each'
	from e2e_test.rb:84:in `run'
	from e2e_test.rb:209:in `<main>'
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

You have an idea what that could be?

**Comment by alexfarrill** [comment]
btw here is how you can run that e2e test: `yarn e2e --only=standalone`

**Comment by IanEdington** [comment]
Great thanks. It's probably due to prop-types not being a dependency. I'll look into it tomorrow and run the e2e tests before re-submitting. Thanks :D

**Comment by IanEdington** [comment]
I identified the issue and wrote it up in https://github.com/Faire/mjml-react/pull/7

Once that's merged and published I'll re-open a PR.

Thanks for helping me find this issue 😄

**Comment by psugihara** [comment]
Makes sense, looks like a good fix, thanks. Excited to merge this. another win for the e2e tests ::)On Sep 27, 2022, at 7:52 PM, Ian Edington ***@***.***> wrote:﻿
I identified the issue and wrote it up in Faire/mjml-react#7
Once that's merged and published I'll re-open a PR.
Thanks for helping me find this issue 😄

**Comment by alexfarrill** [comment]
Nice!  Thanks @IanEdington 🏆

**Comment by IanEdington** [comment]
Another win for your e2e tests. A different test found this issues: https://github.com/Faire/mjml-react/pull/8

**Pull request merged**